api/lib/config.js

```diff
@@ -89,6 +89,7 @@ module.exports = (function() {
     },
 
     features: {
+      dayBeforeImproving: process.env.DAY_BEFORE_IMPROVING,
```

Comment:
Je m'attendrais à ce que `config.js` aille au bout de sa responsabilité de convertir les variables d'environnement en valeurs de paramétrage :
```suggestion
      dayBeforeImproving: (process.env.DAY_BEFORE_IMPROVING, 10) || 4,
```

Reply:
En appliquant cette recommandation, on se retrouve confronté au fait que 
``` javascript
var myNumber = parseInt('0') || 20; // 20
```
Une solution trouvée serait 
```javascript 
function getNumber(number, defaultNumber) {
    return isNaN(parseInt(number, 10)) ? defaultNumber : parseInt(number, 10);
}
```
Je trouve ça moyennement satisfaisant parce que ça ajoute une forme de complexité / de mystère. 
Ça nous indique aussi que la solution courante marche grâce à une pirouette. 
Preneur d'alternatives

Reply:
Très bonne remarque, le cas du zéro nous interdit d'utiliser une écriture compacte avec `||`. Et nous rappelle que la conversion d'une chaîne de configuration en un nombre utilisable, en appliquant une valeur par défaut, n'est pas l'opération triviale qu'on pourrait espérer.

Raison de plus pour que cette responsabilité de conversion soit assurée par un composant en particulier, et non distribuée comme c'est le cas actuellement.

Pourquoi est-ce que `DAY_BEFORE_IMPROVING=0` fonctionne actuellement, alors qu'on a la ligne ci-dessous qui utilise `||` ?

```
MINIMUM_DELAY_IN_DAYS_BEFORE_IMPROVING: settings.features.dayBeforeImproving || 4,
```

Ce qui se passe est que `settings.features.dayBeforeImproving` contient la chaîne `"0"`, qui est donc retenue par l'opérateur `||`. Puis le `parseInt` dans `improvement-service.js` convertit ce `"0"` en `0`. Mais si on n'avait pas spécifié `DAY_BEFORE_IMPROVING` dans l'environnement, on aurait `MINIMUM_DELAY_IN_DAYS_BEFORE_IMPROVING` qui contiendrait le nombre `4`.

On a donc `MINIMUM_DELAY_IN_DAYS_BEFORE_IMPROVING` qui est soit de type chaîne, soit nombre suivant les circonstances… alors que l'interprétation de données provenant de l'extérieur devrait être cantonnée aux contours de l'application.

Je trouve donc la fonction `getNumber` ci-dessous pertinente à ajouter et appeler dans `config.js`. Je ne crois pas qu'il s'agisse d'une solution partielle, il me semble qu'elle fonctionnera normalement même si la valeur par défaut est `0`. Je suggère d'éviter dans son implémentation d'appeler deux fois `parseInt`, et de nommer le paramètre `number` plutôt `string` ou `numberAsString` mais l'idée est là.

Reply:
À défaut, le changement minimal qui reproduit le comportement actuel mais en remontant correctement le `parseInt` au point de contact avec l'extérieur me semble être :

```suggestion
      dayBeforeImproving: parseInt(process.env.DAY_BEFORE_IMPROVING || '4', 10),
```

(par ailleurs je me rends compte que le mot `parseInt` manquait dans ma première suggestion…)